web/js/codeworld-mode.js

```diff
@@ -170,19 +252,21 @@ CodeMirror.defineMode('codeworld', (_config, modeConfig) => {
             return {
                 func: normal,
                 commentLevel: 0,
-                brackets: []
+                layoutContext: [],
+                scanState: NORMAL
             };
         },
 
         token: (stream, state) => {
             const t = state.func(stream, state);
-            if (['variable', 'variable-2'].indexOf(t) !== -1) {
-                const w = stream.current();
-                if (wellKnownWords.hasOwnProperty(w)) return wellKnownWords[w];
+            const w = stream.current();
+            if (wellKnownWords.hasOwnProperty(w)) {
+                if (t.indexOf('variable') !== -1) t.pop(t.indexOf('variable'));
```

Comment:
I don't think Array.pop takes an argument.  I think you can do `t.splice(t.indexOf(...), 1)`  (note that I'm using `splice`, not `slice` which looks similar, but does something different)

Reply:
Thanks!